bedita-app/models/tree.php

```diff
@@ -242,39 +242,70 @@ public function getParents($id = null, $area_id=null, $status = array(), $useCac
      *
      * Possible values are:
      * - `area_id` to limit parents to some publication.
-     * - `status` to limit parents with specific status
+     * - `status` to limit parents with specific status.
+     * - `mantain_hidden_branch` if hidden branches are to mantain. Default true.
      *
-     * @param integer $id
-     * @param array $options
-     * @param array $destination
+     * @param int $id The object id
+     * @param array|int $destination An array of parents or a parent.
+     * @param array $options Options.
+     * @return void
+     * @throws BeditaException If $id is an area or a section with too much destination.
+     * @throws BeditaBadRequestException If errors happen moving a section.
      */
     public function updateTree($id, $destination, array $options = array()) {
         $options += array(
             'area_id' => null,
             'status' => array(),
-			'mantain_hidden_branch' => true
+            'mantain_hidden_branch' => true
         );
         if (!is_array($destination)) {
             $destination = (empty($destination))? array() : array($destination);
         }
-		$excludeFromTreeIds = Configure::read('excludeFromTreeIds');
-		if ($options['mantain_hidden_branch'] === true && !empty($excludeFromTreeIds)) {
-			foreach ($excludeFromTreeIds as $excludeFromTreeId) {
-				$result = $this->find('list', array(
-					'fields' => array('parent_id'),
-					'conditions' => array(
-						'id' => $id,
-						'object_path LIKE' => '%/' . $excludeFromTreeId . '/%'
-					)
-				));
-				if (!empty($result)) {
-					$destination = array_merge($destination, $result);
-				}
-			}
-		}
+
+        $objectTypeId = ClassRegistry::init('BEObject')->field('object_type_id', array('BEObject.id' => $id));
+        if ($objectTypeId === Configure::read('objectTypes.area.id')) {
+            throw new BeditaBadRequestException(__('Publication cannot be moved to destination', true));
+        }
+
+        $isSection = $objectTypeId === Configure::read('objectTypes.section.id');
```

Comment:
[minor] Is strict comparison (`===`) safe?

Reply:
:scream: goot point! It isn't... I fix it.